src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/tests/latest/test_resource_id_deletes.py

```diff
@@ -21,10 +20,15 @@ def test_delete_dependent_resources(self, resource_group):
                  .format(vm_name, resource_group, tag_name, username, password, 'password'))
 
         rsrc_list = self.cmd('resource list --tag {} --query [].id'.format(tag_name)).get_output_in_json()
-
         self.cmd('resource delete --ids {}'.format(' '.join(rsrc_list)))
-        self.cmd('resource list --tag {}'.format(tag_name), checks=[JMESPathCheck('length([])', 0)])
 
+        # to deal with latency caused by caching at ARM frontdoor, we loop till they are all gone
+        for _ in range(30):
+            time.sleep(10)
+            leftovers = self.cmd('resource list --tag {}'.format(tag_name)).get_output_in_json()
+            if not leftovers:
+                return
+        self.fail('Failed to delete all resources as we are still seeing a few leftovers:' + str(leftovers))
```

Comment:
Maybe it would make sense to expose `az resource wait` and leverage that in this test? If the resource continues to appear after the delete call returns then that would be problematic in a scripting scenario as well. Using `az resource wait ... --deleted` would alleviate that.